### DISCUSSION TITLE
quarto publish html book loses layout without embeded resources

### DISCUSSION THREAD

**Discussion by rressler** [quarto-dev/quarto-cli#4055] [Q&A] [answered]
Have a multi-chapter book that looks fine in preview and in the HTML files with embedded resources. However, It is too large to publish on quarto pub as multiple HTML files are way over 100MB with embedded resources. When I render with embedded resource: True I run into this issue as well. https://github.com/quarto-dev/quarto-cli/issues/2462

When I delete embed-resources: true from _quarto.yaml, it will render and publish. The content is there but all the book layout is not.
![Screenshot 2023-01-22 at 7 38 21 PM](https://user-images.githubusercontent.com/59033325/213949276-8e54305d-0400-4a13-b102-0f976a72aaee.png)

It does publish on netlify though.

I looked at the sample _quarto.yaml for R4DS and it looks quite similar.
What am I missing?

**Comment by cderv** [comment]
Can you look into your browser console windows to see the error messages ? (Right Click and Inspect)
It seems resources are not found where they should. 

`embed-resources` is not true by default for websites and books, and there should be a resource folder with all that is needed. When you publish with `quarto publish`, those resources should have been published too. 

Do you see this folder locally ?

**Comment by rressler** [comment]
Christophe,
Thanks for the quick response. You are correct that the resource folder is missing. I looked in both Chrome and Firefox to confirm this was not a browser issue.
[Graphical user interface, application  Description automatically generated]

There is no resource folder in either the top level directory or in the _book directory.

Appreciate any suggestions for next steps.
Thanks,
RIchard
From: Christophe Dervieux ***@***.***>
Date: Monday, January 23, 2023 at 5:51 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] quarto publish html book loses layout without embeded resources (Discussion #4055)

External Email: Use caution with links and attachments.

Can you look into your browser console windows to see the error messages ? (Right Click and Inspect)
It seems resources are not found where they should.

embed-resources is not true by default for websites and books, and there should be a resource folder with all that is needed. When you publish with quarto publish, those resources should have been published too.

Do you see this folder locally ?

**Reply by cderv** [reply]
> Appreciate any suggestions for next steps.

Can you share a repo of your project ? This would help see the config and try on my side

**Comment by rressler** [comment]
Christophe,
Added you in triage role  on GitHub
Thanks!
RIchard

From: Christophe Dervieux ***@***.***>
Date: Monday, January 23, 2023 at 9:37 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] quarto publish html book loses layout without embeded resources (Discussion #4055)

External Email: Use caution with links and attachments.

Appreciate any suggestions for next steps.

Can you share a repo of your project ? This would help see the config and try on my side

**Comment by rressler** [comment]
Christophe
I tried adding you as a collaborator to the repo again. Are you able to see the invite?  Appreciate any suggestions as to how to ensure the libs directory is created in _book. It is creating the images folder but not the libs with the css etc..
Thanks,
Richard

From: Christophe Dervieux ***@***.***>
Date: Monday, January 23, 2023 at 9:37 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] quarto publish html book loses layout without embeded resources (Discussion #4055)

External Email: Use caution with links and attachments.

Appreciate any suggestions for next steps.

Can you share a repo of your project ? This would help see the config and try on my side

**Reply by cderv** [reply]
Yes sorry. I was out. I accepted and will have a look soon. Thanks for reaching out again !

**Comment by rressler** [comment]
Hope it was for fun and not illness. Thanks for the assist!
RIchard

From: Christophe Dervieux ***@***.***>
Date: Monday, February 6, 2023 at 10:13 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] quarto publish html book loses layout without embeded resources (Discussion #4055)

External Email: Use caution with links and attachments.

Yes sorry. I was out. I accepted and will have a look soon. Thanks for reaching out again !

**Reply by cderv** [reply]
Hi Sorry for the delay. 

I can't really run your repo because there are some includes to qmd file that are not in the repo. So can't try for myself. 

I found the website online and could have a look to the deployed version. It seems somehow the `site_libs` was not deployed to quarto pub? 

Do you have this folder locally ? 

Maybe you could try again after updating quarto and cleaning the output to keep only your source and start again ? 

Sorry I am not of much help here - not being able to reproduce makes it hard

**Comment by rressler** [comment, marked as answer]
Christophe,
It is strange in that the embed-resources: true was generating a site_libs but then deleting it when it moved all the html to the _book directory at the end. Perhaps that is because it errored out during the final stages of the render.

embed-resources: was not generating the site_libs directory.

So I followed your suggestion and cleaned out all cache and freeze files and the .quarto directory etc to get to a bare bones folder with just the source files and data and images and CSS, etc..
Then when I rendered again, it worked!!. Something somewhere was causing the problem of not creating the site_libs folder so it could not be moved to _book. I wish I knew what it was.

Now I can quarto render with embed-resources: false and everything works like it should. I can push to both quarto-pub and netlify much faster with the much smaller files.

Thanks for your help!!
Richard

From: Christophe Dervieux ***@***.***>
Date: Friday, February 17, 2023 at 4:59 PM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] quarto publish html book loses layout without embeded resources (Discussion #4055)

External Email: Use caution with links and attachments.

Hi Sorry for the delay.

I can't really run your repo because there are some includes to qmd file that are not in the repo. So can't try for myself.

I found the website online and could have a look to the deployed version. It seems somehow the site_libs was not deployed to quarto pub?

Do you have this folder locally ?

Maybe you could try again after updating quarto and cleaning the output to keep only your source and start again ?

Sorry I am not of much help here - not being able to reproduce makes it hard

**Reply by cderv** [reply]
So cool ! Great ! 

I am really glad it works ! Nice website by the way! 

And sorry again for the delay
